excellent/functions/functions.go

```diff
@@ -174,13 +181,29 @@ func Number(env utils.Environment, value types.XValue) types.XValue {
 //   @(datetime("NOT DATE")) -> ERROR
 //
 // @function datetime(text)
-func DateTime(env utils.Environment, str types.XText) types.XValue {
-	date, err := utils.DateFromString(env, str.Native(), false)
+func DateTime(env utils.Environment, value types.XValue) types.XValue {
+	d, err := types.ToXDateTime(env, value)
 	if err != nil {
 		return types.NewXError(err)
 	}
+	return d
+}
 
-	return types.NewXDateTime(date)
+// Time tries to convert `value` to a time.
+//
+// An error is returned if the value can't be converted.
+//
+//   @(time("10:30")) -> 10:30:00.000000
+//   @(time(datetime("1979-07-18T10:30:45.123456Z"))) -> 10:30:45.123456
+//   @(time("what?")) -> ERROR
```

Comment:
Looks like `10:30pm` will work here, so should probably call out that explicitly as an example.